### PR TITLE
Image card with CTA component - #8069

### PR DESCRIPTION
**Pull request by DipakShrestha18**

### CODE REVIEW

File: src/components/ImageCardWithCTA/ImageCardWithCTA.test.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// import React from 'react';
```
Comment (creido-welly) [inline src/components/ImageCardWithCTA/ImageCardWithCTA.test.tsx]: Is this commented out for a reason?

Comment (DipakShrestha18) [reply src/components/ImageCardWithCTA/ImageCardWithCTA.test.tsx]: No, forgot to push it.

Comment (DipakShrestha18) [reply src/components/ImageCardWithCTA/ImageCardWithCTA.test.tsx]: @creido-welly ready for re-review.

File: src/components/ImageCardWithCTA/ImageCardWithCTA.tsx [hunk 2]
```diff
@@ -0,0 +1,92 @@
+import React from 'react';
```
Comment (creido-welly) [inline src/components/ImageCardWithCTA/ImageCardWithCTA.tsx:1]: There is a lot of duplication here, understandably so, but from a React point of view we should try to make use of _composition_ - or in other words make use of props children to create a base component which has all of the common stuff and into which the markup which is different can be nested.

e.g.

```jsx
<CardBase>
  {description && <RichText>{description}</RichText>}
</CardBase>
```

This is an oversimplification but illustrates the basics. More than happy to chat through this if you like.

Longer term we want to overhaul how listings and cards are composed as it's pretty complex and therefore not ideal. Updating both this component and ImageCard to use the same CardBase will be good step in the right direction and certainly reduce duplication.

Some more info on composition:
https://reactjs.org/docs/composition-vs-inheritance.html
https://epicreact.dev/one-react-mistake-thats-slowing-you-down/

**Review by creido-welly — CHANGES_REQUESTED** [review]
There is a lot of duplication here, understandably so, but from a React point of view we should try to make use of _composition_ - or in other words make use of props children to create a base component which has all of the common stuff and into which the markup which is different can be nested.

e.g.

```jsx
<CardBase>
  {description && <RichText>{description}</RichText>}
</CardBase>
```

This is an oversimplification but illustrates the basics. More than happy to chat through this if you like.

Longer term we want to overhaul how listings and cards are composed as it's pretty complex and therefore not ideal. Updating both this component and ImageCard to use the same CardBase will be good step in the right direction and certainly reduce duplication.

Some more info on composition:
https://reactjs.org/docs/composition-vs-inheritance.html
https://epicreact.dev/one-react-mistake-thats-slowing-you-down/

**Comment by chriddell** [comment]
@creido-welly How about if ImageCard became the "base", and then we had ImageCardWithMetaData and ImageCardWithCTALink as the "extensions"... 🤯

**Comment by creido-welly** [comment]
> @creido-welly How about if ImageCard became the "base", and then we had ImageCardWithMetaData and ImageCardWithCTALink as the "extensions"...

Sure if it makes sense to do that. ImageCard as it stands is effectively ImageCardWithMetaData.

**Comment by DipakShrestha18** [comment]
> > @creido-welly How about if ImageCard became the "base", and then we had ImageCardWithMetaData and ImageCardWithCTALink as the "extensions"...
> 
> Sure if it makes sense to do that. ImageCard as it stands is effectively ImageCardWithMetaData.

Do you mean extending the existing component?

**Comment by chriddell** [comment]
Easiest way to think about it is probably:

- Renaming ImageCard -> ImageCardWithMetaData
- Copying all of the code from ImageCardWithMetaData to a new component: ImageCard
- Refactoring ImageCard to:
  - remove the parts which are unique to ImageCardWithMetaData and ImageCardWithCTALink 
  - add a `children` prop
- Refactoring ImageCardWithMetaData and ImageCardWithCTALink to use ImageCard and passing unique parts as `children`


Basically we would have something similar to*:

```js
const ImageCard = ({ children, ...otherProps }) => (
  <article className="c-image-card">
    <figure className="c-image-card__img">
      <img src={src} />
    </figure>
    {children}
  </article>
);

const ImageCardWithMetaData = ({ props }) => (
  <ImageCard>
    <div className="c-image-card__meta">
      <p>{props.date}</p>
      <p>{props.author}</p>
    </div>
  </ImageCard>
);

const ImageCardWithCTALink = ({ props }) => (
  <ImageCard>
    <a className="c-image-card__cta" href={props.href}>
  </ImageCard>
);
```

*this is VERY simplified

**Comment by DipakShrestha18** [comment]
Should it make sense to work that as separate job? It feels like we're side tracking as Yssy wants to get this component out this sprint. What's your view?

I can create a ticket for this and work on it next week.

**Comment by creido-welly** [comment]
@DipakShrestha18 since time is short I have no objection as long as it does get done

**Review by creido-welly — APPROVED** [review]

**Comment by DipakShrestha18** [comment]
I have create github ticket for refactoring and consistent naming. 
https://github.com/wellcometrust/corporate/issues/8108

**Pull request merged**